query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -35,14 +38,83 @@
   // while the selectedColumns
   // is a set of column names.
   private final LinkedHashSet<Expression> allSelections;
+  private final Map<String,String> timeSeriesColumnMap;
+  private final Map<String,List<Long>> timeFilterMap;
 
   public ExecutionContext(String tenantId, QueryRequest request) {
     this.tenantId = tenantId;
     this.selectedColumns = new LinkedHashSet<>();
     this.allSelections = new LinkedHashSet<>();
+    this.timeSeriesColumnMap = new HashMap<>();
```

Comment:
There's some tech debt in this class, but for our new code, since this is being passed around we should be using only immutable collections.

Reply:
+1 make sense.